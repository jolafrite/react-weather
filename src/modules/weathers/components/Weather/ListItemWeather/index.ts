export * from './ListItemWeather';
export { default } from './ListItemWeather';
