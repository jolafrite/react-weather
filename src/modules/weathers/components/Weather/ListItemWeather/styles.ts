import { makeStyles, createStyles } from "@material-ui/core";
import { rem } from "polished";

export const useStyles = makeStyles(
  createStyles({
    weatherIcon: {},
    location: {
      display: 'flex',
      flexDirection: 'row',
      alignItems: 'center',
    },
    city: {
      fontSize: rem(12),
      marginRight: rem(2),
    },
    country: {
      fontSize: rem(12),
    },
    temp: {
      fontSize: rem(30),
      fontWeight: 'bold',
    },
  })
);
