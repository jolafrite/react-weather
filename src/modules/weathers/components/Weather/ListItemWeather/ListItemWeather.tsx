import React, { FC, memo } from 'react';
import { Avatar, ListItemAvatar, ListItemText, Typography } from '@material-ui/core';
import { useStyles } from './styles';
import { IWeather } from '../../../models';

export interface IListItemWeatherProps {
  city: string;
  country: string;
  icon: string;
  temp: number;
}

const ListItemWeather: FC<IListItemWeatherProps> = props => {
  const classes = useStyles();
  const { city, country, icon, temp } = props;

  return (
    <>
      <ListItemAvatar>
        <Avatar
          className={classes.weatherIcon}
          src={`https://openweathermap.org/img/w/${icon}.png`}
        />
      </ListItemAvatar>
      <ListItemText
        primary={
          <div className={classes.location}>
            <Typography>{city}, </Typography>
            <Typography>{country}</Typography>
          </div>
        }
        secondary={
          <Typography className={classes.temp}>{Math.round(temp)}Â°</Typography>
        }
      />
    </>
  )
};

export default memo(ListItemWeather);
