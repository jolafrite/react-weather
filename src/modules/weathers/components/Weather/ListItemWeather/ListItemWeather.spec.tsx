import React from 'react';
import { cleanup, render } from "@testing-library/react";
import ListItemWeather, { IListItemWeatherProps } from "./ListItemWeather";

describe('component: LoadingView', () => {

  const createComponent = (values = {}) => {
    const props: IListItemWeatherProps = {
      ...values,
      city: 'Hong Kong',
      country: 'CN',
      icon: '04d',
      temp: 30,
    };

    return render(
      <ListItemWeather
        {...props}
      />
    );
  };

  afterEach(cleanup);

  test('default render', () => {
    const component = createComponent();

    expect(
      component.container
    ).toMatchSnapshot();
  });

});
