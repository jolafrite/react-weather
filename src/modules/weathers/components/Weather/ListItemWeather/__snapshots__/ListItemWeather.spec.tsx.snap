// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component: LoadingView default render 1`] = `
<div>
  <div
    class="MuiListItemAvatar-root"
  >
    <div
      class="MuiAvatar-root makeStyles-weatherIcon-1"
    >
      <img
        class="MuiAvatar-img"
        src="https://openweathermap.org/img/w/04d.png"
      />
    </div>
  </div>
  <div
    class="MuiListItemText-root MuiListItemText-multiline"
  >
    <span
      class="MuiTypography-root MuiListItemText-primary MuiTypography-body1"
    >
      <div
        class="makeStyles-location-2"
      >
        <p
          class="MuiTypography-root MuiTypography-body1"
        >
          Hong Kong
          , 
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1"
        >
          CN
        </p>
      </div>
    </span>
    <p
      class="MuiTypography-root makeStyles-temp-5 MuiTypography-body1"
    >
      30
      Â°
    </p>
  </div>
</div>
`;
